extensions/python/PythonScriptFlowFile.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utility>
+#include <memory>
+#include <string>
+
+#include "core/FlowFile.h"
+
+#include "PythonScriptFlowFile.h"
+
+namespace org::apache::nifi::minifi::extensions::python {
+
+PythonScriptFlowFile::PythonScriptFlowFile(std::shared_ptr<core::FlowFile> flow_file)
+    : flow_file_(std::move(flow_file)) {
+}
+
+std::string PythonScriptFlowFile::getAttribute(const std::string &key) {
+  if (!flow_file_) {
+    throw std::runtime_error("Access of FlowFile after it has been released");
+  }
+
+  std::string value;
+  flow_file_->getAttribute(key, value);
+  return value;
+}
+
+bool PythonScriptFlowFile::addAttribute(const std::string &key, const std::string &value) {
+  if (!flow_file_) {
+    throw std::runtime_error("Access of FlowFile after it has been released");
+  }
+
+  return flow_file_->addAttribute(key, value);
+}
+
+bool PythonScriptFlowFile::setAttribute(const std::string &key, const std::string &value) {
+  if (!flow_file_) {
+    throw std::runtime_error("Access of FlowFile after it has been released");
+  }
+
+  return flow_file_->setAttribute(key, value);
+}
+
+bool PythonScriptFlowFile::updateAttribute(std::string key, std::string value) {
+  if (!flow_file_) {
+    throw std::runtime_error("Access of FlowFile after it has been released");
+  }
+
+  return flow_file_->updateAttribute(std::move(key), std::move(value));
+}
+
+bool PythonScriptFlowFile::removeAttribute(std::string key) {
+  if (!flow_file_) {
+    throw std::runtime_error("Access of FlowFile after it has been released");
+  }
+
+  return flow_file_->removeAttribute(std::move(key));
+}
+
+std::shared_ptr<core::FlowFile> PythonScriptFlowFile::getFlowFile() {
+  return flow_file_;
+}
+
+void PythonScriptFlowFile::releaseFlowFile() {
+  flow_file_.reset();
```

Comment:
this method (contrary to `LuaScriptFlowFile`) seems to be unused, if this is meant to be, we could replace the checks with `gsl_Expect`

Reply:
I've removed this whole class in https://github.com/apache/nifi-minifi-cpp/pull/1504/commits/7a9008d5e23eb41a1d4eaca6045fcf3889ea4ffe